src/components/Search/Search.tsx

```diff
@@ -19,26 +22,23 @@ interface SearchProps {
 
 // OR store can be global
 
-export const Search: FC<SearchProps> = props => {
-  const [searchedValue, setSearchedValue] = useState('');
-
-  const onChange = value => {
-    console.log(value); // for debugging
-    setSearchedValue(value);
+export function Search(props: SearchProps) {
+  const onChange = (value) => {
+    props.dispatch({ type: SET_SEARCH, payload: value });
```

Comment:
When we are talking about user's search, it would be better not to invoke business logic on each symbol. We should wait for the user to stop typing and search one time after this. :)

Could you pls try to improve this functionality using https://www.npmjs.com/package/lodash.debounce or any other implementation https://codepen.io/jpkempf/pen/MbePGB ?